pkg/infra/filestorage/config.go

```diff
@@ -0,0 +1,49 @@
+package filestorage
+
+type BackendType string
+
+const (
+	BackendTypeFS BackendType = "fs"
+	BackendTypeDB BackendType = "db"
+)
+
+type fsBackendConfig struct {
+	RootPath string `json:"path"`
+}
+
+type backendConfig struct {
+	Type                BackendType `json:"type"`
+	Name                string      `json:"name"`
+	AllowedPrefixes     []string    `json:"allowedPrefixes"`     // null -> all paths are allowed
+	SupportedOperations []Operation `json:"supportedOperations"` // null -> all operations are supported
+
+	FSBackendConfig *fsBackendConfig `json:"fsBackendConfig"`
```

Comment:
for ts integration, it will be easier if the property name matches exactly the type. 

Although many things will be "fs" based, I think we will want/need more specific ones like:
* BackendLocalDiskConfig
* BackendGCSConfig
* BackendS3Config
* BackendGitConfig   (with special subconfig for github/gitlab?)